internal/api/registry/manifests_test.go

```diff
@@ -662,3 +662,91 @@ func TestImageManifestWrongBlobSize(t *testing.T) {
 		}.Check(t, h)
 	})
 }
+
+func TestListManifestRequiredLabels(t *testing.T) {
+	testWithPrimary(t, nil, func(s test.Setup) {
+		h := s.Handler
+		token := s.GetToken(t, "repository:test1/foo:pull,push")
+
+		baseImageConfig := test.BaseImageConfig
+		config, ok := baseImageConfig["config"].(map[string]interface{})
+		if ok {
+			config["Labels"] = map[string]string{"foo": "is there", "bar": "is there"}
+		}
+		baseImageConfig["config"] = config
+
+		image1 := test.GenerateImage(test.GenerateExampleLayer(1))
+		image2 := test.GenerateImageFromConfig(baseImageConfig, test.GenerateExampleLayer(2))
+		// s.Clock.Step()
+		image1.MustUpload(t, s, fooRepoRef, "first")
+		// s.Clock.Step()
+		image2.MustUpload(t, s, fooRepoRef, "second")
+		// s.Clock.Step()
+
+		//setup required labels on account for failure
+		_, err := s.DB.Exec(
+			`UPDATE accounts SET required_labels = $1 WHERE name = $2`,
+			"foo,somethingelse,andalsothis", "test1",
+		)
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		// cannot upload image list manifest with no labels
+		list1 := test.GenerateImageList(image1)
+		assert.HTTPRequest{
+			Method: "PUT",
+			Path:   "/v2/test1/foo/manifests/" + list1.Manifest.Digest.String(),
+			Header: map[string]string{
+				"Authorization": "Bearer " + token,
+				"Content-Type":  list1.Manifest.MediaType,
+			},
+			Body:         assert.ByteData(list1.Manifest.Contents),
+			ExpectStatus: http.StatusBadRequest,
+			ExpectBody: assert.JSONObject{
+				"errors": []assert.JSONObject{{
+					"code":    "MANIFEST_INVALID",
+					"message": "missing required labels: foo,somethingelse,andalsothis",
+					"detail":  nil,
+				}},
+			},
+		}.Check(t, h)
+
+		// image list manifest push with labels should succeed
+		list2 := test.GenerateImageList(image2)
+		assert.HTTPRequest{
+			Method: "PUT",
+			Path:   "/v2/test1/foo/manifests/latest",
+			Header: map[string]string{
+				"Authorization": "Bearer " + token,
+				"Content-Type":  "application/vnd.docker.distribution.manifest.v2+json",
```

Comment:
You need a different MIME type here:
```suggestion
				"Content-Type":  "application/vnd.docker.distribution.manifest.list.v2+json",
```